genesyscloud/resource_genesyscloud_knowledge_knowledgebase.go

```diff
@@ -20,23 +20,64 @@ func getAllKnowledgeKnowledgebases(_ context.Context, clientConfig *platformclie
 	resources := make(ResourceIDMetaMap)
 	knowledgeAPI := platformclientv2.NewKnowledgeApiWithConfig(clientConfig)
 
-	for pageNum := 1; ; pageNum++ {
-		const pageSize = 100
-		knowledgeBases, _, getErr := knowledgeAPI.GetKnowledgeKnowledgebases("", "", "", fmt.Sprintf("%v", pageSize), "", "", false, "", "")
+	publishedEntities, err := getAllKnowledgebaseEntities(*knowledgeAPI, true)
+	if err != nil {
+		return nil, diag.Errorf("%v", err)
+	}
+
+	for _, knowledgeBase := range *publishedEntities {
+		resources[*knowledgeBase.Id] = &ResourceMeta{Name: *knowledgeBase.Name}
+	}
+
+	unpublishedEntities, err := getAllKnowledgebaseEntities(*knowledgeAPI, false)
+	if err != nil {
+		return nil, diag.Errorf("%v", err)
+	}
+
+	for _, knowledgeBase := range *unpublishedEntities {
+		resources[*knowledgeBase.Id] = &ResourceMeta{Name: *knowledgeBase.Name}
+	}
+
+	return resources, nil
+}
+
+func getAllKnowledgebaseEntities(knowledgeApi platformclientv2.KnowledgeApi, published bool) (*[]platformclientv2.Knowledgebase, error) {
+	var (
+		after    string
+		entities []platformclientv2.Knowledgebase
+	)
+
+	const pageSize = 100
+	for i := 0; ; i++ {
+		knowledgeBases, _, getErr := knowledgeApi.GetKnowledgeKnowledgebases("", after, "", fmt.Sprintf("%v", pageSize), "", "", published, "", "")
 		if getErr != nil {
-			return nil, diag.Errorf("Failed to get page of knowledge bases: %v", getErr)
+			return nil, fmt.Errorf("Failed to get page of knowledge bases: %v", getErr)
```

Comment:
Hey Charlie why did you use a diag error vs a a go error?  Not questioning the decision just seeking to understand the issue.

Reply:
No reason. Just changed it to diag, definitely makes more sense